pkg/agent/agent.go

```diff
@@ -99,6 +99,21 @@ func (i *Initializer) setupOVSBridge() error {
 	}
 	i.nodeConfig.BridgeName = i.ovsBridgeClient.GetBridgeName()
 
+	if err := i.prepareOVSBridge(); err != nil {
+		// prepareOVSBridge only works on windows platform. The operation have chance to fail on the first time agent
+		// starts up when OVS bridge uplink and local inteface has not been configured. If the operation fails, the
+		// host will can not communicate with external network. To make sure the agent can connect to API server in
+		// next retry, this step delete OVS bridge and HNS network created previously which will restore the
+		// host network.
+		if err := i.ovsBridgeClient.Delete(); err != nil {
```

Comment:
Since these logics are very specific to Windows, I feel better to move them (line 102 to 113) to prepareOVSBridge(), then code in agent.go is easier to understand.
If bridge deletion  is a concern, we can add a comment here saying prepareOVSBridge() in Windows could delete bridge at failure.

What you think?

Reply:
Or another way:
1. Move line 117 - 132 to Initialize().
2. Move setupOVSBridge() to agent_linux.go and agent_windows.go, including ovsBridgeClient.Create(). It is fine to duplicate a single func call in both _linux.go and _windows.go.

Reply:
Thanks Jianjun, I move the failure handling codes in prepareOVSBridge as defer function.